### DISCUSSION TITLE
microTVM or zephyr bugs, No SOURCES given to Zephyr library

### DISCUSSION THREAD

**Discussion by wangpengabc** [zephyrproject-rtos/zephyr#52080] [General] [answered]
After the configuration of microTVM in [TVM](https://github.com/apache/tvm) and zephyr in my local environment, I run the "python micro_aot.py" in tvm/gallery/how_to/work_with_microtvm subdirectory. The following errors occur.

```
zephyrproject/zephyr/subsys/random/CMakeLists.txt:11 (message):
  

      Warning: CONFIG_TEST_RANDOM_GENERATOR is not a truly random generator.
      This capability is not secure and it is provided for testing purposes only.
      Use it carefully.


CMake Warning at /home/sdc/wangpeng/dlc/zephyr/zephyrproject/zephyr/CMakeLists.txt:824 (message):
  No SOURCES given to Zephyr library: drivers__console

  Excluding target from build.
```

```
tmp/tmp7tvfh3o8/project/src/main.c:31:10: fatal error: drivers/gpio.h: No such file or directory
   31 | #include <drivers/gpio.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by wangpengabc** [comment]
Same as [#13303](https://github.com/apache/tvm/issues/13303) in TVM

**Comment by laurenmurphyx64** [comment]
Closed bug since we believe this is a bug on microTVM's side, but you're welcome to discuss here.

**Comment by stephanosio** [comment, marked as answer]
This seems to be a bug on the microTVM side.

> No SOURCES given to Zephyr library: drivers__console

It looks like you are not selecting any console drivers.

> tmp/tmp7tvfh3o8/project/src/main.c:31:10: fatal error: drivers/gpio.h: No such file or directory

You need to add `zephyr/` prefix to the Zephyr headers now.

```c
#include <zephyr/drivers/gpio.h>
```
